apps/autoscheduler/test.cpp

```diff
@@ -0,0 +1,471 @@
+#include "Halide.h"
+
+#include <dlfcn.h>
+
+using namespace Halide;
+
+int main(int argc, char **argv) {
+    if (!dlopen("libauto_schedule.so", RTLD_LAZY)) {
+        std::cerr << "Failed to load autoscheduler: " << dlerror() << "\n";
+        return 1;
+    }
+
+    MachineParams params(32, 16000000, 40);
+    // Use a fixed target for the analysis to get consistent results from this test.
+    Target target("x86-64-linux-sse41-avx-avx2");
+
+    Var x("x"), y("y");
+
+    if (1) {
+        // In a point-wise pipeline, everything should be fully fused.
```

Comment:
This is called a "test" but does anything actually verify the output? (ie is it really just for human inspection?)

Reply:
There's no output verification. It's more a test that the autoscheduler doesn't crash or take the age of the universe.